src/pkg/util/oci/helper.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package oci
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+	"github.com/sylabs/sif/pkg/sif"
+)
+
+const (
+	// ConfigSpec holds the name for the OCI runtime spec file
+	ConfigSpec = "config.json"
+)
+
+// LoadConfigSpec loads the config.json oci runtime spec
+// from the provided path to a SIF.
+func LoadConfigSpec(Path string) (spec *specs.Spec, err error) {
+	// load the SIF (singularity image file)
+	fimg, err := sif.LoadContainer(Path, false)
+	if err != nil {
+		sylog.Fatalf("Error loading SIF %s:\t%s", Path, err)
+	}
+
+	// lookup of a descriptor of type DataGenericJSON
+	descr := sif.Descriptor{
+		Datatype: sif.DataGenericJSON,
+	}
+	d, match, _ := fimg.GetFromDescr(descr)
```

Comment:
See comments above about how to understand and use GetFromDescr() (what descr fields to pass in and how to interpret the return values)